src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExecuteActionTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Threading.Tasks;
+using Xunit;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using System.Reflection;
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using System.Threading;
+using System;
+using System.IO;
+using System.Diagnostics;
+using Microsoft.Diagnostics.Tools.Monitor;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class ExecuteActionTests
+    {
+        private const int TokenTimeoutMs = 2000; // Must be less than DelayMs in the ExecuteApp
+
+        private const string TaskCanceledMessage = "A task was canceled";
+
+        [Fact]
+        public async Task ExecuteAction_ZeroExitCode()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            options.Path = DotNetHost.HostExePath;
+            options.Arguments = GenerateArgumentsString(new string[] { "ZeroExitCode" });
+
+            CollectionRuleActionResult result = await action.ExecuteAsync(options, null, CreateCancellationToken());
+
+            Assert.Equal("0", result.OutputValues["ExitCode"]);
+        }
+
+        [Fact]
+        public async Task ExecuteAction_NonzeroExitCode()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            options.Path = DotNetHost.HostExePath;
+            options.Arguments = GenerateArgumentsString(new string[] { "NonzeroExitCode" });
+
+            InvalidOperationException invalidOperationException = await Assert.ThrowsAsync<InvalidOperationException>(
+                () => action.ExecuteAsync(options, null, CreateCancellationToken()));
+
+            Assert.Contains(string.Format(Strings.ErrorMessage_NonzeroExitCode, "-1"), invalidOperationException.Message);
+        }
+
+        [Fact]
+        public async Task ExecuteAction_TokenCancellation()
+        {
+            ExecuteAction action = new();
+
+            ExecuteOptions options = new();
+
+            options.Path = DotNetHost.HostExePath;
+            options.Arguments = GenerateArgumentsString(new string[] { "TokenCancellation" }); ;
+
+            TaskCanceledException taskCanceledException = await Assert.ThrowsAsync<TaskCanceledException>(
+                () => action.ExecuteAsync(options, null, CreateCancellationToken()));
+
+            Assert.Contains(TaskCanceledMessage, taskCanceledException.Message);
```

Comment:
This is not necessary. Testing that it is throwing TaskCanceledException is sufficient.